### PR TITLE
gh-116349: Deprecate `platform.java_ver` function

### PR DESCRIPTION
**Pull request by sobolevn**
* Issue: gh-116349

----
📚 Documentation preview 📚: https://cpython-previews--116471.org.readthedocs.build/

### CODE REVIEW

File: Doc/library/platform.rst [hunk 1]
```diff
@@ -196,6 +196,9 @@ Java Platform
    ``(os_name, os_version, os_arch)``. Values which cannot be determined are set to
    the defaults given as parameters (which all default to ``''``).
 
+   .. deprecated-removed:: 3.13 3.15
+      It was only useful for Jython support.
```
Comment (AlexWaygood) [inline Doc/library/platform.rst]: Maybe we could be slightly more expansive here (and ditto for the whatsnew entry and the NEWS entry)

```suggestion
      It was largely untested, had a confusing API,
      and was only useful for Jython support.
```

**Review by AlexWaygood — APPROVED** [review]
LGTM, just a minor nit about the docs

File: Doc/whatsnew/3.13.rst [hunk 2]
```diff
@@ -973,6 +973,10 @@ Pending Removal in Python 3.15
   They will be removed in Python 3.15.
   (Contributed by Victor Stinner in :gh:`105096`.)
 
+* :func:`platform.java_ver` is deprecated and will be removed in 3.15.
```
Comment (vstinner) [inline Doc/whatsnew/3.13.rst:980]: You have to repeat the deprecation in the Deprecation section.

File: Lib/platform.py [hunk 3]
```diff
@@ -503,7 +503,7 @@ def mac_ver(release='', versioninfo=('', '', ''), machine=''):
     return release, versioninfo, machine
 
 def _java_getprop(name, default):
-
+    """This protected helper is deprecated in 3.13 and will be removed in 3.15"""
```
Comment (vstinner) [inline Lib/platform.py]:
```suggestion
    """This private helper is deprecated in 3.13 and will be removed in 3.15"""
```

**Review by hugovk — APPROVED** [review]

**Review by vstinner — APPROVED** [review]
LGTM

**Comment by sobolevn** [comment]
Thanks everyone! ❤️

**Review by Scutua — APPROVED** [review]

**Pull request merged**